src/main/scala/uk/gov/hmrc/cache/repository/CacheRepository.scala

```diff
@@ -16,33 +16,53 @@
 
 package uk.gov.hmrc.cache.repository
 
+import javax.inject.Inject
 import play.api.libs.json._
-import play.modules.reactivemongo.MongoDbConnection
+import play.modules.reactivemongo.{MongoDbConnection, ReactiveMongoComponent}
 import reactivemongo.api.DB
 import reactivemongo.api.commands.LastError
 import reactivemongo.bson._
 import reactivemongo.play.json.commands.{DefaultJSONCommandError, JSONFindAndModifyCommand}
+import uk.gov.hmrc.cache.TimeToLive
 import uk.gov.hmrc.cache.model.{Cache, Id}
 import uk.gov.hmrc.mongo._
 import uk.gov.hmrc.mongo.json.ReactiveMongoFormats
 
 import scala.concurrent.{ExecutionContext, Future}
 
 
-trait CacheRepository extends ReactiveRepository[Cache, Id] with UniqueIndexViolationRecovery {
-
-  def createOrUpdate(id: Id, key: String, toCache: JsValue): Future[DatabaseUpdate[Cache]] = ???
-
-  protected def saveOrUpdate(findQuery: => Future[Option[Cache]], ifNotFound: => Future[Cache], modifiers: (Cache) => Cache)(implicit ec: ExecutionContext): Future[DatabaseUpdate[Cache]] = ???
+trait CacheRepository extends ReactiveRepository[Cache, Id] {
+  def createOrUpdate[T](id: Id, key: String, toCache: T)(implicit writes: Writes[T]): Future[DatabaseUpdate[Cache]]
 }
 
-@deprecated("Please use injected CacheMongoRepository to your class", since = "6.x")
+@deprecated("Please use injected CacheRepositoryFactory for CacheRepository creation", since = "6.x")
 object CacheRepository extends MongoDbConnection {
+  @deprecated("Please use injected CacheRepositoryFactory to CacheRepository creation", since = "6.x")
```

Comment:
7.x

Reply:
this deprecation would be same as above object - in most cases people did not use object but rather apply on it and for example intliJ doesn't show deprecation in this case.